phone_verify/docs/how_to_integrate.rst

```diff
@@ -0,0 +1,177 @@
+    from phone_verify.api import VerificationViewSet
+
+    default_router = DefaultRouter(trailing_slash=False)
+    default_router.register('phone', VerificationViewSet, basename='phone')
+
+    urlpatterns = default_router.urls
+
+.. note:: It is recommended that you should verify the phone number of users before their registration.
+
+Case 2: Verify phone number at the time of user registration
+************************************************************
+
+This is the case when you choose to integrate your user registration process with OTP verification.
+
+It can be achieved by overriding the default `verify` view of `phone_verify.api.VerificationViewSet`. Following a code example to achieve the same.
+
+.. note:: Here, you'll first register a phone number using `/api/phone/register` and then will use the modified functionality of endpoint `/api/phone/verify` to create a user on successfull verification of the phone number.
+
+- Add a default router in `api_urls` to redirect on one of your custom viewset:
+
+.. code-block:: python
+
+    # In api_urls.py
+
+    from rest_framework.routers import DefaultRouter
+    from yourapp.api import YourCustomViewSet
+
+    default_router = DefaultRouter(trailing_slash=False)
+
+    default_router.register('phone', YourCustomViewSet, basename='phone')
+
+    urlpatterns = default_router.urls
+
+
+- Create YourCustomSerializer:
+
+.. code-block:: python
+
+    # In serializers.py
+
+    from rest_framework import serializers
+
+    from phone_verify.serializers import SMSVerificationSerializer
+
+    class YourUserSerializer(serializers.Serializer):
+        username = serializers.CharField(required=True)
+        email = serializers.EmailField(required=True)
+        password = serializers.CharField(required=True)
+        first_name = serializers.CharField(default="First")
+        ...
+
+    class YourCustomSerializer(UserSerializer, SMSVerificationSerializer):
+        ...
+
+
+- Add a service to create users
+
+.. code-block:: python
+
+    # In your services.py
+
+    from django.contrib.auth import get_user_model
+
+    def create_user_account(username, email, password, **extra_args):
+        user = get_user_model().objects.create_user(
+            username=username, email=email, password=password, **extra_args
+        )
+        ...
+        return user
+
+
+- Create YourCustomViewSet:
+
+.. code-block:: python
+
+    # In your api.py
+
+    from rest_framework.decorators import action
+    from rest_framework.permissions import AllowAny
+    from rest_framework.response import Response
+
+    from phone_verify.api import VerificationViewSet
+    from phone_verify import serializers as phone_serializers
+
+    from . import services, serializers
+
+
+    class YourCustomViewSet(VerificationViewSet):
+
+        @action(detail=False, methods=['POST'], permission_classes=[AllowAny], serializer_class=serializers.YourCustomSerializer)
+        def verify(self, request):
```

Comment:
We can name this `verify_and_register` and then mention in a docstring that most of the code is corresponding to the `verify` view already present in the package.